imageregistry/v1/types.go

```diff
@@ -303,8 +302,65 @@ type ImageRegistryConfigStorageAzure struct {
 	// object.
 	// +optional
 	CloudName string `json:"cloudName,omitempty"`
+	// networkAccess defines the network access properties for the storage account.
+	// Defaults to type: External.
+	// +kubebuilder:default={"type": "External"}
+	// +optional
+	NetworkAccess *AzureNetworkAccess `json:"networkAccess,omitempty"`
+}
+
+type AzureNetworkAccess struct {
+	// type is the network access level to be used for the storage account.
+	// type: Internal means the storage account will be private, type: External
+	// means the storage account will be publicly accessible.
+	// When type: Internal is used, a vnetName, subNetName and privateEndpointName
+	// may optionally be specified. If unspecificed, the image registry operator
+	// will discover vnet and subnet names, and generate a privateEndpointName.
+	// Defaults to "External".
+	// +kubebuilder:default:="External"
+	Type AzureNetworkAccessType `json:"type,omitempty"`
```

Comment:
Should have an explicit `+optional` tag on this field

Reply:
Done